lib/gems/pending/util/miq-ipmi.rb

```diff
@@ -185,10 +185,11 @@ def run_command(ipmi_cmd, *args)
     command_args[nil]    = args unless args.empty?
 
     begin
-      return MiqUtil.runcmd("ipmitool", :params => command_args)
-    rescue => err
-      return err.to_s if continue_on_error == true && $?.exitstatus == 1
-      raise "Command:<#{command_line}> exited with status:<#{$?.exitstatus}>\nCommand output:\n#{err}"
+      AwesomeSpawn.run!("ipmitool", :params => command_args, :combined_output => true).output
+    rescue AwesomeSpawn::CommandResultError => err
+      return err.to_s if continue_on_error == true && err.exit_status == 1
+
+      raise "Command:<#{err.command_line}> exited with status:<#{err.exit_status}>\nCommand output:\n#{err}"
```

Comment:
Also note this whole error handling was a bunch of bugs beforehand. `command_line` standalone is an non-existent variable name, and `$?` was never set by `runcmd`